test.js

```diff
@@ -74,7 +74,7 @@ describe('the server', function() {
     server = cproc.spawn('node', ['ass-stubs/server-test.js', port]);
     var isDone = false;
     server.stdout.on('data', function(data) {
-      if (!isDone) { done(); isDone = true; }
+      if (data.toString().indexOf('ready') >= 0 && !isDone) { done(); isDone = true; }
```

Comment:
This protects against the subsequent tests starting before the server is ready.  What seemed to be happening was that when the 'ass-stubs/server-test.js' was spawned, this would pass immediately after it saw the "http://127.0.0.1:80/try.html" line that gets printed out at the beginning of server.js - this happens before the server is started, causing subsequent tests to fail with 'ECONNREFUSED' when they tried to hit the test server before it started.